src/templates/helpers/templateHelpers.ts

```diff
@@ -1,10 +1,174 @@
-import { mongoidPrimitiveTypes } from '../../model/dataTypes/primitiveDataTypes';
+import _ from "lodash";
+import Handlebars, { HelperOptions } from "handlebars";
+import { reservedWords } from "../../constants";
+import {
+  mongoidPrimitiveTypes,
+  mongoosePrimitiveTypes,
+  typeScriptPrimitiveTypes,
+} from "../../model/dataTypes/primitiveDataTypes";
+import MemberVariable from "../../model/dataTypes/MemberVariable";
+import DataType from "../../model/dataTypes/DataType";
+import { jsonChoiceName } from "../../utils";
+
+export function trimInterfaceName(typeName: string): string {
+  if (typeName.startsWith("IPrimitive")) {
+    return typeName.slice(10);
+  }
+  if (typeName.startsWith("Primitive")) {
+    return typeName.slice(9);
+  }
+  if (typeName.startsWith("I")) {
+    return typeName.slice(1);
+  }
+  return typeName;
+}
+
+export function getTypeScriptPrimitive(type: string): string {
+  return typeScriptPrimitiveTypes[_.lowerFirst(type)];
+}
+
+export function getTypeScriptInterfacePrimitive(type: string): string {
+  const trimmedName = trimInterfaceName(type);
+  return typeScriptPrimitiveTypes[_.lowerFirst(trimmedName)];
+}
 
 export function getMongoidPrimitive(type: string): string {
-  return mongoidPrimitiveTypes[type];
+  return mongoidPrimitiveTypes[_.lowerFirst(type)];
+}
+
+Handlebars.registerHelper("getMongoidPrimitive", getMongoidPrimitive);
+
+Handlebars.registerHelper("getTypeScriptPrimitive", getTypeScriptPrimitive);
+
+Handlebars.registerHelper("isReservedKeyword", function isReserved(
+  this: Handlebars.TemplateDelegate<string>,
+  token: string,
+  options: HelperOptions
+): string {
+  if (reservedWords.includes(token)) {
+    return options.fn(this);
```

Comment:
Codacy found an issue: [Unsafe call of an any typed value.](https://app.codacy.com/gh/MeasureAuthoringTool/model-info-parser/pullRequest?prid=5966907)